plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoSession.java

```diff
@@ -640,6 +645,28 @@ else if (value instanceof Document) {
                 typeSignature = new TypeSignature(StandardTypes.ROW, parameters);
             }
         }
+        else if (value instanceof DBRef) {
+            List<TypeSignatureParameter> parameters = new ArrayList<>();
+
+            Optional<TypeSignature> idFieldType = guessFieldType(((DBRef) value).getId());
+            if (idFieldType.isPresent()) {
+                parameters.add(TypeSignatureParameter.namedTypeParameter(new NamedTypeSignature(Optional.of(new RowFieldName(ID)), idFieldType.get())));
+            }
```

Comment:
Let's throw an exception if failed to guess field type. DBRef without `id` field doesn't make sense. 
Thus, we can make DBRef's field size always 3 and simplify `MongoPageSource` logic.

Reply:
`checkState(3 == type.getTypeParameters().size(), "fieldName doesn't match with type size : %s", type);`

Does this look like a good way to throw the error ?

Reply:
I mean we should throw an exception if `idFieldType` is empty.

Reply:
I pasted the wrong code, my bad.
`checkState(idFieldType.isPresent(), “Could not resolve type for $id field!”);`

what I wanted to ask is that is checkState the correct way to throw here or should I add an explicit new form of exception?

Reply:
I would implement like this:
```java
            DBRef dbRef = (DBRef) value;
            TypeSignature idFieldType = guessFieldType(dbRef.getId())
                    .orElseThrow(() -> new UnsupportedOperationException("Unable to guess id field type of DBRef from: " + dbRef.getId()));
```

Reply:
Got it.